src/shared/file-glob.js

```diff
@@ -23,42 +24,17 @@ import { last } from './array-util.js';
  * - '**' match everything
  *
  * @param {string} glob a fs-like glob to test paths with
- * @returns {function(string): boolean}
+ * @returns {function(string, boolean): boolean}
  */
 export function compileGlob(glob) {
-  const expr = new RegExp(
-    `^(${glob
-      .replace(/(?<=^|[{,])[^{},]+(?=$|[,{}])/g, match =>
-        match
-          .split(/[\\/]/g)
-          .reduce(
-            (carry, part) =>
-              carry.length ? [...carry, `${last(carry)}/`, `${last(carry)}/${part}`] : [part],
-            []
-          )
-          .join(',')
-      )
-      .replace(/\{|\}|,|\.|\/|\*{1,2}/g, match =>
-        match === '{'
-          ? '('
-          : match === '}'
-          ? ')'
-          : match === ','
-          ? '|'
-          : match === '.'
-          ? '\\.'
-          : match === '/'
-          ? '[\\\\/]'
-          : match === '*'
-          ? '[^\\\\/]+'
-          : '.*'
-      )})$`
-  );
-
-  return target => expr.test(target);
+  const mm = new Minimatch(glob);
+  return (path, partial = false) => {
+    const result = mm.match(path, partial);
```

Comment:
[The second parameter to the `match` method is undocumented.](https://www.npmjs.com/package/minimatch#methods), so its use would make this project vulnerable to regressions in future releases of minimatch. I think we'll have to forgo the memoization and construct a new object for each "match" calculation.

Separately, it wouldn't hurt to petition the maintainers of minimatch to commit to that API, but I wouldn't gate this patch on the resolution of such a proposal (though if you do file an issue, referencing it in a comment would certainly be helpful).

Reply:
Nice catch - I switched to caching 2 matchers - wastes a little bit of time in the test suite, but the main use for this it will still save a bunch of re-compiling the glob

Reply:
Great idea!